src/library/scala/collection/IterableOnce.scala

```diff
@@ -743,56 +765,91 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
    *  overridden for more efficient traversal orders.
    *
-   *  $undefinedorder
    *  $willNotTerminateInf
    *
-   *  @tparam A1     a type parameter for the binary operator, a supertype of `A`.
-   *  @param z       a neutral element for the fold operation; may be added to the result
-   *                 an arbitrary number of times, and must not change the result (e.g., `Nil` for list concatenation,
-   *                 0 for addition, or 1 for multiplication).
-   *  @param op      a binary operator that must be associative.
-   *  @return        the result of applying the fold operator `op` between all the elements and `z`, or `z` if this $coll is empty.
+   *  @tparam A1     The type parameter for the binary operator, a supertype of `A`.
+   *  @param z       An initial value; may be used an arbitrary number of times in the
+   *                 computation of the result; must be a neutral element for `op` for the
+   *                 result to always be the same across runs.
+   *  @param op      A binary operator; must be associative for the result to always be the
+   *                 same across runs.
+   *  @return        The result of applying `op` between all the elements and `z`, or `z`
+   *                 if this $coll is empty.
    */
   def fold[A1 >: A](z: A1)(op: (A1, A1) => A1): A1 = foldLeft(z)(op)
 
-  /** Reduces the elements of this $coll using the specified associative binary operator.
+  /** Applies the given binary operator to all elements of this $coll.
    *
-   *  $undefinedorder
+   *  For each application of the operator, each operand is either an element of this
+   *  $coll or another such application of the operator. $undefinedOrder $exactlyOnce
    *
-   *  @tparam B      A type parameter for the binary operator, a supertype of `A`.
-   *  @param op       A binary operator that must be associative.
-   *  @return         The result of applying reduce operator `op` between all the elements if the $coll is nonempty.
-   *  @throws UnsupportedOperationException if this $coll is empty.
+   *  If this collection is ordered, then for any application of the operator, the
+   *  element(s) appearing in the left operand will precede those in the right.
+   *
+   *  $orderIndependentReduce
+   *  $willNotTerminateInf
+   *
+   *  @tparam B      The type parameter for the binary operator, a supertype of `A`.
+   *  @param op      A binary operator; must be associative for the result to always be the
+   *                 same across runs.
+   *  @return        The result of applying `op` between all the elements if the $coll is
+   *                 nonempty.
+   *  @throws        UnsupportedOperationException if this $coll is empty.
    */
   def reduce[B >: A](op: (B, B) => B): B = reduceLeft(op)
 
-  /** Reduces the elements of this $coll, if any, using the specified
-   *  associative binary operator.
+  /** Optionally reduces this $coll with the given binary operator.
```

Comment:
```suggestion
  /** If this $coll is nonempty, reduces it with the given binary operator.
```

Reply:
`Optionally` is the usual, what's the word, objective or impersonal adverb, like "hopefully". You mean you're full of hope? You mean you've got options?

It is explained for the uninitiated on the next line.